library/Requests/Transport/fsockopen.php

```diff
@@ -160,8 +160,9 @@ public function request($url, $headers = array(), $data = array(), $options = ar
 				$request_body = $data;
 			}
 
-			if (!empty($data)) {
+			if (!empty($data) || $options['type'] === Requests::POST) {
 				if (!isset($case_insensitive_headers['Content-Length'])) {
+					// Prevent 411 errors from some servers when the body is empty.
```

Comment:
The comment is in the wrong spot, as it actually explains the extra condition, not the header string.
```suggestion
			// Always include Content-length on POST requests to prevent
			// 411 errors from some servers when the body is empty.
			if (!empty($data) || $options['type'] === Requests::POST) {
				if (!isset($case_insensitive_headers['Content-Length'])) {
```

Reply:
Fixed up